buildNpmPackage: tries to install axios but it's not in the npmDeps causing `ENOTCACHED`

### Describe the bug
A clear and concise description of what the bug is.

```
Validating consistency between /private/tmp/nix-build-appium-2.1.3.drv-0/source/package-lock.json and /nix/store/vlwmmbrinlmpcq0d7829v1bdr8j39ypg-appium-2.1.3-npm-deps/package-lock.json
Installing dependencies
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/axios failed: cache mode is 'only-if-cached' but no cached response is available.

npm ERR! Log files were not written due to an error writing to the directory: /nix/store/vlwmmbrinlmpcq0d7829v1bdr8j39ypg-appium-2.1.3-npm-deps/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to write to the logs directory -- look above that for the actual error.
2. Set `npmFlags = [ "--legacy-peer-deps" ]`
```

Grep confirms the dep is preset in both lock files but not in `npm-deps/_cacache/index-v5`.

### Steps To Reproduce

```Nix
{ buildNpmPackage, fetchFromGitHub, lib }:

buildNpmPackage rec {
  pname = "appium";
  version = "2.1.3";

  src = fetchFromGitHub {
    owner = pname;
    repo = pname;
    rev = "${pname}@${version}";
    hash = "sha256-+UE6XK9gxPRm5eRl5RXQnWEjkBD7jwaw/o+4TmiJhIA=";
  };

  npmDepsHash = "sha256-4ji2We7DzduaT3FMHUqhWDyXj8QlvQvc2D91ATl9yd4=";
}
```

### Notify maintainers

@winterqt @lilyinstarlight

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

e56990880811a451abd32515698c712788be5720

So I tried regenerating the lock files using `npm install --package-lock-only` but it attempts and fails to build the entire project when I do that? Wtf?

Anyhow, I tried the master branch where I can rebuild the lockfile and this is the error now:

```
Validating consistency between /private/tmp/nix-build-appium-2.1.3.drv-0/source/package-lock.json and /nix/store/vbf37p84ibgxfl09wqa2ph67hwpc1gas-appium-2.1.3-npm-deps/package-lock.json
Installing dependencies
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: chai@4.3.7
npm WARN node_modules/chai
npm WARN   dev chai@"4.3.8" from the root project
npm WARN   3 more (chai-as-promised, mocha-chai-jest-snapshot, sinon-chai)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer chai@">= 2.1.2 < 5" from chai-as-promised@7.1.1
npm WARN node_modules/chai-as-promised
npm WARN   dev chai-as-promised@"7.1.1" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: chai@4.3.7
npm WARN node_modules/chai
npm WARN   dev chai@"4.3.8" from the root project
npm WARN   3 more (chai-as-promised, mocha-chai-jest-snapshot, sinon-chai)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer chai@"^4.3.6" from mocha-chai-jest-snapshot@1.1.4
npm WARN node_modules/mocha-chai-jest-snapshot
npm WARN   dev mocha-chai-jest-snapshot@"1.1.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: chai@4.3.7
npm WARN node_modules/chai
npm WARN   dev chai@"4.3.8" from the root project
npm WARN   3 more (chai-as-promised, mocha-chai-jest-snapshot, sinon-chai)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer chai@"^4.0.0" from sinon-chai@3.7.0
npm WARN node_modules/sinon-chai
npm WARN   dev sinon-chai@"3.7.0" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@">=7.0.0" from eslint-config-prettier@8.10.0
npm WARN node_modules/eslint-config-prettier
npm WARN   dev eslint-config-prettier@"9.0.0" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@">=7.0.0" from eslint-config-prettier@8.10.0
npm WARN node_modules/eslint-config-prettier
npm WARN   dev eslint-config-prettier@"9.0.0" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^7 || ^8.2.0" from eslint-find-rules@4.1.0
npm WARN node_modules/eslint-find-rules
npm WARN   dev eslint-find-rules@"4.1.0" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.28.1
npm WARN node_modules/eslint-plugin-import
npm WARN   dev eslint-plugin-import@"2.28.1" from the root project
npm WARN   3 more (eslint-import-resolver-typescript, ...)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.28.1
npm WARN node_modules/eslint-plugin-import
npm WARN   dev eslint-plugin-import@"2.28.1" from the root project
npm WARN   3 more (eslint-import-resolver-typescript, ...)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.28.1
npm WARN node_modules/eslint-plugin-import
npm WARN   dev eslint-plugin-import@"2.28.1" from the root project
npm WARN   3 more (eslint-import-resolver-typescript, ...)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.28.1
npm WARN node_modules/eslint-plugin-import
npm WARN   dev eslint-plugin-import@"2.28.1" from the root project
npm WARN   3 more (eslint-import-resolver-typescript, ...)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8" from eslint-plugin-import@2.28.1
npm WARN node_modules/eslint-plugin-import
npm WARN   dev eslint-plugin-import@"2.28.1" from the root project
npm WARN   3 more (eslint-import-resolver-typescript, ...)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@">=7.0.0" from eslint-plugin-mocha@10.1.0
npm WARN node_modules/eslint-plugin-mocha
npm WARN   dev eslint-plugin-mocha@"10.1.0" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@">=7.0.0" from eslint-plugin-mocha@10.1.0
npm WARN node_modules/eslint-plugin-mocha
npm WARN   dev eslint-plugin-mocha@"10.1.0" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@">=7.0.0" from eslint-plugin-mocha@10.1.0
npm WARN node_modules/eslint-plugin-mocha
npm WARN   dev eslint-plugin-mocha@"10.1.0" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^7.0.0 || ^8.0.0" from eslint-plugin-promise@6.1.1
npm WARN node_modules/eslint-plugin-promise
npm WARN   dev eslint-plugin-promise@"6.1.1" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^7.0.0 || ^8.0.0" from eslint-plugin-promise@6.1.1
npm WARN node_modules/eslint-plugin-promise
npm WARN   dev eslint-plugin-promise@"6.1.1" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: appium-monorepo@0.0.0
npm WARN Found: eslint@8.47.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"8.48.0" from the root project
npm WARN   10 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^7.0.0 || ^8.0.0" from eslint-plugin-promise@6.1.1
npm WARN node_modules/eslint-plugin-promise
npm WARN   dev eslint-plugin-promise@"6.1.1" from the root project
npm WARN   2 more (@appium/eslint-config-appium, @appium/eslint-config-appium-ts)
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/@tsconfig%2fnode14 failed: cache mode is 'only-if-cached' but no cached response is available.

npm ERR! Log files were not written due to an error writing to the directory: /nix/store/vbf37p84ibgxfl09wqa2ph67hwpc1gas-appium-2.1.3-npm-deps/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to write to the logs directory -- look above that for the actual error.
2. Set `npmFlags = [ "--legacy-peer-deps" ]`

```

Thanks.

Updated the lockfile and injected it into the source and now I get the next unfetched dep:

```
Installing dependencies
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/supports-color failed: cache mode is 'only-if-cached' but no cached response is available.

npm ERR! Log files were not written due to an error writing to the directory: /nix/store/a66v20j5zayn05lfz99v317j5xi6wak0-appium-2.1.3-npm-deps/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to write to the logs directory -- look above that for the actual error.
2. Set `npmFlags = [ "--legacy-peer-deps" ]`
```

I have a related issue — a package that doesn't show up in the lockfile at all is being installed, causing an ENOTCACHED error.

<details>
  <summary>error logs</summary>

```console
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-spago-0.93.16-npm-deps/package-lock.json
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/npm
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-sp
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-spago-0.93.16-npm->
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodej>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-s>
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-spago-0.93.16-npm>
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-node>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm->
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-spago-0.93>
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p3adyi85bm67gpik>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfjisj53l3hisp27j2k>
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp27j2ksh>
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfji>
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.93.16.tgz
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store/xbiakh1rfjisj53l3hisp2>
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiak>
npm verb title npm ci
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/a5gk49kgw4m481z1f8j70jgi6q3v0pai-spago-0.9>
source root is package
setting SOURCE_DATE_EPOCH to timestamp 499162500 of file package/package.json
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
Executing npmConfigHook
Configuring npm
Validating consistency between /build/package/package-lock.json and /nix/store>
Installing dependencies
npm verb cli /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/bin/node /nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodej>
npm info using npm@9.6.7
npm info using node@v18.17.1
npm verb logfile could not create logs-dir: Error: EACCES: permission denied, mkdir '/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-s>
npm verb title npm ci
npm verb argv "ci" "--ignore-scripts" "--loglevel" "verbose"
npm verb logfile logs-max:10 dir:/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm-spago-0.93.16-npm-deps/_logs/2023-10-20T00_59_00_684Z-
npm verb logfile could not be created: Error: ENOENT: no such file or directory, open '/nix/store/xbiakh1rfjisj53l3hisp27j2kshqmfm>
npm verb logfile no logfile created
npm verb stack Error: request to https://registry.npmjs.org/env-paths failed: cache mode is 'only-if-cached' but no cached respons>
npm verb stack     at cacheFetch (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/node_modules/mak>
npm verb stack     at async fetch (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/node_modules/ma>
npm verb stack     at async RegistryFetcher.packument (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules>
npm verb stack     at async RegistryFetcher.manifest (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/>
npm verb stack     at async #nodeFromEdge (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/node_mo>
npm verb stack     at async #buildDepStep (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/node_mo>
npm verb stack     at async Arborist.buildIdealTree (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/n>
npm verb stack     at async CI.exec (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/lib/commands/>
npm verb stack     at async module.exports (/nix/store/8w46c2hf7hclc5p3adyi85bm67gpikqf-nodejs-18.17.1/lib/node_modules/npm/lib/cl>
npm verb cwd /build/package
npm verb Linux 6.4.8
npm verb node v18.17.1
npm verb npm  v9.6.7
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/env-paths failed: cache mode is>
npm verb exit 1
npm verb code 1

npm ERR! Log files were not written due to an error writing to the directory: >
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs i>

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to writ>
2. Set `npmFlags = [ "--legacy-peer-deps" ]`
```
</details>

Notably, env-paths doesn't show up in the package-lock.json file at all.

<details>
  <summary>package-lock.json</summary>

```json
{
  "name": "spago",
  "lockfileVersion": 3,
  "requires": true,
  "packages": {
    "": {
      "dependencies": {
        "better-sqlite3": "^8.7.0"
      }
    },
    "node_modules/base64-js": {
      "version": "1.5.1",
      "resolved": "https://registry.npmjs.org/base64-js/-/base64-js-1.5.1.tgz",
      "integrity": "sha512-AKpaYlHn8t4SVbOHCy+b5+KKgvR4vrsD8vbvrbiQJps7fKDTkjkDry6ji0rUJjC0kzbNePLwzxq8iypo41qeWA==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ]
    },
    "node_modules/better-sqlite3": {
      "version": "8.7.0",
      "resolved": "https://registry.npmjs.org/better-sqlite3/-/better-sqlite3-8.7.0.tgz",
      "integrity": "sha512-99jZU4le+f3G6aIl6PmmV0cxUIWqKieHxsiF7G34CVFiE+/UabpYqkU0NJIkY/96mQKikHeBjtR27vFfs5JpEw==",
      "hasInstallScript": true,
      "dependencies": {
        "bindings": "^1.5.0",
        "prebuild-install": "^7.1.1"
      }
    },
    "node_modules/bindings": {
      "version": "1.5.0",
      "resolved": "https://registry.npmjs.org/bindings/-/bindings-1.5.0.tgz",
      "integrity": "sha512-p2q/t/mhvuOj/UeLlV6566GD/guowlr0hHxClI0W9m7MWYkL1F0hLo+0Aexs9HSPCtR1SXQ0TD3MMKrXZajbiQ==",
      "dependencies": {
        "file-uri-to-path": "1.0.0"
      }
    },
    "node_modules/bl": {
      "version": "4.1.0",
      "resolved": "https://registry.npmjs.org/bl/-/bl-4.1.0.tgz",
      "integrity": "sha512-1W07cM9gS6DcLperZfFSj+bWLtaPGSOHWhPiGzXmvVJbRLdG82sH/Kn8EtW1VqWVA54AKf2h5k5BbnIbwF3h6w==",
      "dependencies": {
        "buffer": "^5.5.0",
        "inherits": "^2.0.4",
        "readable-stream": "^3.4.0"
      }
    },
    "node_modules/buffer": {
      "version": "5.7.1",
      "resolved": "https://registry.npmjs.org/buffer/-/buffer-5.7.1.tgz",
      "integrity": "sha512-EHcyIPBQ4BSGlvjB16k5KgAJ27CIsHY/2JBmCRReo48y9rQ3MaUzWX3KVlBa4U7MyX02HdVj0K7C3WaB3ju7FQ==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ],
      "dependencies": {
        "base64-js": "^1.3.1",
        "ieee754": "^1.1.13"
      }
    },
    "node_modules/chownr": {
      "version": "1.1.4",
      "resolved": "https://registry.npmjs.org/chownr/-/chownr-1.1.4.tgz",
      "integrity": "sha512-jJ0bqzaylmJtVnNgzTeSOs8DPavpbYgEr/b0YL8/2GO3xJEhInFmhKMUnEJQjZumK7KXGFhUy89PrsJWlakBVg=="
    },
    "node_modules/decompress-response": {
      "version": "6.0.0",
      "resolved": "https://registry.npmjs.org/decompress-response/-/decompress-response-6.0.0.tgz",
      "integrity": "sha512-aW35yZM6Bb/4oJlZncMH2LCoZtJXTRxES17vE3hoRiowU2kWHaJKFkSBDnDR+cm9J+9QhXmREyIfv0pji9ejCQ==",
      "dependencies": {
        "mimic-response": "^3.1.0"
      },
      "engines": {
        "node": ">=10"
      },
      "funding": {
        "url": "https://github.com/sponsors/sindresorhus"
      }
    },
    "node_modules/deep-extend": {
      "version": "0.6.0",
      "resolved": "https://registry.npmjs.org/deep-extend/-/deep-extend-0.6.0.tgz",
      "integrity": "sha512-LOHxIOaPYdHlJRtCQfDIVZtfw/ufM8+rVj649RIHzcm/vGwQRXFt6OPqIFWsm2XEMrNIEtWR64sY1LEKD2vAOA==",
      "engines": {
        "node": ">=4.0.0"
      }
    },
    "node_modules/detect-libc": {
      "version": "2.0.2",
      "resolved": "https://registry.npmjs.org/detect-libc/-/detect-libc-2.0.2.tgz",
      "integrity": "sha512-UX6sGumvvqSaXgdKGUsgZWqcUyIXZ/vZTrlRT/iobiKhGL0zL4d3osHj3uqllWJK+i+sixDS/3COVEOFbupFyw==",
      "engines": {
        "node": ">=8"
      }
    },
    "node_modules/end-of-stream": {
      "version": "1.4.4",
      "resolved": "https://registry.npmjs.org/end-of-stream/-/end-of-stream-1.4.4.tgz",
      "integrity": "sha512-+uw1inIHVPQoaVuHzRyXd21icM+cnt4CzD5rW+NC1wjOUSTOs+Te7FOv7AhN7vS9x/oIyhLP5PR1H+phQAHu5Q==",
      "dependencies": {
        "once": "^1.4.0"
      }
    },
    "node_modules/expand-template": {
      "version": "2.0.3",
      "resolved": "https://registry.npmjs.org/expand-template/-/expand-template-2.0.3.tgz",
      "integrity": "sha512-XYfuKMvj4O35f/pOXLObndIRvyQ+/+6AhODh+OKWj9S9498pHHn/IMszH+gt0fBCRWMNfk1ZSp5x3AifmnI2vg==",
      "engines": {
        "node": ">=6"
      }
    },
    "node_modules/file-uri-to-path": {
      "version": "1.0.0",
      "resolved": "https://registry.npmjs.org/file-uri-to-path/-/file-uri-to-path-1.0.0.tgz",
      "integrity": "sha512-0Zt+s3L7Vf1biwWZ29aARiVYLx7iMGnEUl9x33fbB/j3jR81u/O2LbqK+Bm1CDSNDKVtJ/YjwY7TUd5SkeLQLw=="
    },
    "node_modules/fs-constants": {
      "version": "1.0.0",
      "resolved": "https://registry.npmjs.org/fs-constants/-/fs-constants-1.0.0.tgz",
      "integrity": "sha512-y6OAwoSIf7FyjMIv94u+b5rdheZEjzR63GTyZJm5qh4Bi+2YgwLCcI/fPFZkL5PSixOt6ZNKm+w+Hfp/Bciwow=="
    },
    "node_modules/github-from-package": {
      "version": "0.0.0",
      "resolved": "https://registry.npmjs.org/github-from-package/-/github-from-package-0.0.0.tgz",
      "integrity": "sha512-SyHy3T1v2NUXn29OsWdxmK6RwHD+vkj3v8en8AOBZ1wBQ/hCAQ5bAQTD02kW4W9tUp/3Qh6J8r9EvntiyCmOOw=="
    },
    "node_modules/ieee754": {
      "version": "1.2.1",
      "resolved": "https://registry.npmjs.org/ieee754/-/ieee754-1.2.1.tgz",
      "integrity": "sha512-dcyqhDvX1C46lXZcVqCpK+FtMRQVdIMN6/Df5js2zouUsqG7I6sFxitIC+7KYK29KdXOLHdu9zL4sFnoVQnqaA==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ]
    },
    "node_modules/inherits": {
      "version": "2.0.4",
      "resolved": "https://registry.npmjs.org/inherits/-/inherits-2.0.4.tgz",
      "integrity": "sha512-k/vGaX4/Yla3WzyMCvTQOXYeIHvqOKtnqBduzTHpzpQZzAskKMhZ2K+EnBiSM9zGSoIFeMpXKxa4dYeZIQqewQ=="
    },
    "node_modules/ini": {
      "version": "1.3.8",
      "resolved": "https://registry.npmjs.org/ini/-/ini-1.3.8.tgz",
      "integrity": "sha512-JV/yugV2uzW5iMRSiZAyDtQd+nxtUnjeLt0acNdw98kKLrvuRVyB80tsREOE7yvGVgalhZ6RNXCmEHkUKBKxew=="
    },
    "node_modules/lru-cache": {
      "version": "6.0.0",
      "resolved": "https://registry.npmjs.org/lru-cache/-/lru-cache-6.0.0.tgz",
      "integrity": "sha512-Jo6dJ04CmSjuznwJSS3pUeWmd/H0ffTlkXXgwZi+eq1UCmqQwCh+eLsYOYCwY991i2Fah4h1BEMCx4qThGbsiA==",
      "dependencies": {
        "yallist": "^4.0.0"
      },
      "engines": {
        "node": ">=10"
      }
    },
    "node_modules/mimic-response": {
      "version": "3.1.0",
      "resolved": "https://registry.npmjs.org/mimic-response/-/mimic-response-3.1.0.tgz",
      "integrity": "sha512-z0yWI+4FDrrweS8Zmt4Ej5HdJmky15+L2e6Wgn3+iK5fWzb6T3fhNFq2+MeTRb064c6Wr4N/wv0DzQTjNzHNGQ==",
      "engines": {
        "node": ">=10"
      },
      "funding": {
        "url": "https://github.com/sponsors/sindresorhus"
      }
    },
    "node_modules/minimist": {
      "version": "1.2.8",
      "resolved": "https://registry.npmjs.org/minimist/-/minimist-1.2.8.tgz",
      "integrity": "sha512-2yyAR8qBkN3YuheJanUpWC5U3bb5osDywNB8RzDVlDwDHbocAJveqqj1u8+SVD7jkWT4yvsHCpWqqWqAxb0zCA==",
      "funding": {
        "url": "https://github.com/sponsors/ljharb"
      }
    },
    "node_modules/mkdirp-classic": {
      "version": "0.5.3",
      "resolved": "https://registry.npmjs.org/mkdirp-classic/-/mkdirp-classic-0.5.3.tgz",
      "integrity": "sha512-gKLcREMhtuZRwRAfqP3RFW+TK4JqApVBtOIftVgjuABpAtpxhPGaDcfvbhNvD0B8iD1oUr/txX35NjcaY6Ns/A=="
    },
    "node_modules/napi-build-utils": {
      "version": "1.0.2",
      "resolved": "https://registry.npmjs.org/napi-build-utils/-/napi-build-utils-1.0.2.tgz",
      "integrity": "sha512-ONmRUqK7zj7DWX0D9ADe03wbwOBZxNAfF20PlGfCWQcD3+/MakShIHrMqx9YwPTfxDdF1zLeL+RGZiR9kGMLdg=="
    },
    "node_modules/node-abi": {
      "version": "3.51.0",
      "resolved": "https://registry.npmjs.org/node-abi/-/node-abi-3.51.0.tgz",
      "integrity": "sha512-SQkEP4hmNWjlniS5zdnfIXTk1x7Ome85RDzHlTbBtzE97Gfwz/Ipw4v/Ryk20DWIy3yCNVLVlGKApCnmvYoJbA==",
      "dependencies": {
        "semver": "^7.3.5"
      },
      "engines": {
        "node": ">=10"
      }
    },
    "node_modules/once": {
      "version": "1.4.0",
      "resolved": "https://registry.npmjs.org/once/-/once-1.4.0.tgz",
      "integrity": "sha512-lNaJgI+2Q5URQBkccEKHTQOPaXdUxnZZElQTZY0MFUAuaEqe1E+Nyvgdz/aIyNi6Z9MzO5dv1H8n58/GELp3+w==",
      "dependencies": {
        "wrappy": "1"
      }
    },
    "node_modules/prebuild-install": {
      "version": "7.1.1",
      "resolved": "https://registry.npmjs.org/prebuild-install/-/prebuild-install-7.1.1.tgz",
      "integrity": "sha512-jAXscXWMcCK8GgCoHOfIr0ODh5ai8mj63L2nWrjuAgXE6tDyYGnx4/8o/rCgU+B4JSyZBKbeZqzhtwtC3ovxjw==",
      "dependencies": {
        "detect-libc": "^2.0.0",
        "expand-template": "^2.0.3",
        "github-from-package": "0.0.0",
        "minimist": "^1.2.3",
        "mkdirp-classic": "^0.5.3",
        "napi-build-utils": "^1.0.1",
        "node-abi": "^3.3.0",
        "pump": "^3.0.0",
        "rc": "^1.2.7",
        "simple-get": "^4.0.0",
        "tar-fs": "^2.0.0",
        "tunnel-agent": "^0.6.0"
      },
      "bin": {
        "prebuild-install": "bin.js"
      },
      "engines": {
        "node": ">=10"
      }
    },
    "node_modules/pump": {
      "version": "3.0.0",
      "resolved": "https://registry.npmjs.org/pump/-/pump-3.0.0.tgz",
      "integrity": "sha512-LwZy+p3SFs1Pytd/jYct4wpv49HiYCqd9Rlc5ZVdk0V+8Yzv6jR5Blk3TRmPL1ft69TxP0IMZGJ+WPFU2BFhww==",
      "dependencies": {
        "end-of-stream": "^1.1.0",
        "once": "^1.3.1"
      }
    },
    "node_modules/rc": {
      "version": "1.2.8",
      "resolved": "https://registry.npmjs.org/rc/-/rc-1.2.8.tgz",
      "integrity": "sha512-y3bGgqKj3QBdxLbLkomlohkvsA8gdAiUQlSBJnBhfn+BPxg4bc62d8TcBW15wavDfgexCgccckhcZvywyQYPOw==",
      "dependencies": {
        "deep-extend": "^0.6.0",
        "ini": "~1.3.0",
        "minimist": "^1.2.0",
        "strip-json-comments": "~2.0.1"
      },
      "bin": {
        "rc": "cli.js"
      }
    },
    "node_modules/readable-stream": {
      "version": "3.6.2",
      "resolved": "https://registry.npmjs.org/readable-stream/-/readable-stream-3.6.2.tgz",
      "integrity": "sha512-9u/sniCrY3D5WdsERHzHE4G2YCXqoG5FTHUiCC4SIbr6XcLZBY05ya9EKjYek9O5xOAwjGq+1JdGBAS7Q9ScoA==",
      "dependencies": {
        "inherits": "^2.0.3",
        "string_decoder": "^1.1.1",
        "util-deprecate": "^1.0.1"
      },
      "engines": {
        "node": ">= 6"
      }
    },
    "node_modules/safe-buffer": {
      "version": "5.2.1",
      "resolved": "https://registry.npmjs.org/safe-buffer/-/safe-buffer-5.2.1.tgz",
      "integrity": "sha512-rp3So07KcdmmKbGvgaNxQSJr7bGVSVk5S9Eq1F+ppbRo70+YeaDxkw5Dd8NPN+GD6bjnYm2VuPuCXmpuYvmCXQ==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ]
    },
    "node_modules/semver": {
      "version": "7.5.4",
      "resolved": "https://registry.npmjs.org/semver/-/semver-7.5.4.tgz",
      "integrity": "sha512-1bCSESV6Pv+i21Hvpxp3Dx+pSD8lIPt8uVjRrxAUt/nbswYc+tK6Y2btiULjd4+fnq15PX+nqQDC7Oft7WkwcA==",
      "dependencies": {
        "lru-cache": "^6.0.0"
      },
      "bin": {
        "semver": "bin/semver.js"
      },
      "engines": {
        "node": ">=10"
      }
    },
    "node_modules/simple-concat": {
      "version": "1.0.1",
      "resolved": "https://registry.npmjs.org/simple-concat/-/simple-concat-1.0.1.tgz",
      "integrity": "sha512-cSFtAPtRhljv69IK0hTVZQ+OfE9nePi/rtJmw5UjHeVyVroEqJXP1sFztKUy1qU+xvz3u/sfYJLa947b7nAN2Q==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ]
    },
    "node_modules/simple-get": {
      "version": "4.0.1",
      "resolved": "https://registry.npmjs.org/simple-get/-/simple-get-4.0.1.tgz",
      "integrity": "sha512-brv7p5WgH0jmQJr1ZDDfKDOSeWWg+OVypG99A/5vYGPqJ6pxiaHLy8nxtFjBA7oMa01ebA9gfh1uMCFqOuXxvA==",
      "funding": [
        {
          "type": "github",
          "url": "https://github.com/sponsors/feross"
        },
        {
          "type": "patreon",
          "url": "https://www.patreon.com/feross"
        },
        {
          "type": "consulting",
          "url": "https://feross.org/support"
        }
      ],
      "dependencies": {
        "decompress-response": "^6.0.0",
        "once": "^1.3.1",
        "simple-concat": "^1.0.0"
      }
    },
    "node_modules/string_decoder": {
      "version": "1.3.0",
      "resolved": "https://registry.npmjs.org/string_decoder/-/string_decoder-1.3.0.tgz",
      "integrity": "sha512-hkRX8U1WjJFd8LsDJ2yQ/wWWxaopEsABU1XfkM8A+j0+85JAGppt16cr1Whg6KIbb4okU6Mql6BOj+uup/wKeA==",
      "dependencies": {
        "safe-buffer": "~5.2.0"
      }
    },
    "node_modules/strip-json-comments": {
      "version": "2.0.1",
      "resolved": "https://registry.npmjs.org/strip-json-comments/-/strip-json-comments-2.0.1.tgz",
      "integrity": "sha512-4gB8na07fecVVkOI6Rs4e7T6NOTki5EmL7TUduTs6bu3EdnSycntVJ4re8kgZA+wx9IueI2Y11bfbgwtzuE0KQ==",
      "engines": {
        "node": ">=0.10.0"
      }
    },
    "node_modules/tar-fs": {
      "version": "2.1.1",
      "resolved": "https://registry.npmjs.org/tar-fs/-/tar-fs-2.1.1.tgz",
      "integrity": "sha512-V0r2Y9scmbDRLCNex/+hYzvp/zyYjvFbHPNgVTKfQvVrb6guiE/fxP+XblDNR011utopbkex2nM4dHNV6GDsng==",
      "dependencies": {
        "chownr": "^1.1.1",
        "mkdirp-classic": "^0.5.2",
        "pump": "^3.0.0",
        "tar-stream": "^2.1.4"
      }
    },
    "node_modules/tar-stream": {
      "version": "2.2.0",
      "resolved": "https://registry.npmjs.org/tar-stream/-/tar-stream-2.2.0.tgz",
      "integrity": "sha512-ujeqbceABgwMZxEJnk2HDY2DlnUZ+9oEcb1KzTVfYHio0UE6dG71n60d8D2I4qNvleWrrXpmjpt7vZeF1LnMZQ==",
      "dependencies": {
        "bl": "^4.0.3",
        "end-of-stream": "^1.4.1",
        "fs-constants": "^1.0.0",
        "inherits": "^2.0.3",
        "readable-stream": "^3.1.1"
      },
      "engines": {
        "node": ">=6"
      }
    },
    "node_modules/tunnel-agent": {
      "version": "0.6.0",
      "resolved": "https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.6.0.tgz",
      "integrity": "sha512-McnNiV1l8RYeY8tBgEpuodCC1mLUdbSN+CYBL7kJsJNInOP8UjDDEwdk6Mw60vdLLrr5NHKZhMAOSrR2NZuQ+w==",
      "dependencies": {
        "safe-buffer": "^5.0.1"
      },
      "engines": {
        "node": "*"
      }
    },
    "node_modules/util-deprecate": {
      "version": "1.0.2",
      "resolved": "https://registry.npmjs.org/util-deprecate/-/util-deprecate-1.0.2.tgz",
      "integrity": "sha512-EPD5q1uXyFxJpCrLnCc1nHnq3gOa6DZBocAIiI2TaSCA7VCJ1UJDMagCzIkXNsUYfD1daK//LTEQ8xiIbrHtcw=="
    },
    "node_modules/wrappy": {
      "version": "1.0.2",
      "resolved": "https://registry.npmjs.org/wrappy/-/wrappy-1.0.2.tgz",
      "integrity": "sha512-l4Sp/DRseor9wL6EvV2+TuQn63dMkPjZ/sp9XkghTEbV9KlPS1xUsZ3u7/IQO4wxtcFB4bgpQPRcR3QCvezPcQ=="
    },
    "node_modules/yallist": {
      "version": "4.0.0",
      "resolved": "https://registry.npmjs.org/yallist/-/yallist-4.0.0.tgz",
      "integrity": "sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A=="
    }
  }
}
```
</details>

--- a/pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh
+++ b/pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh
@@ -14,7 +14,7 @@ npmInstallHook() {
         local dest="$packageOut/$(dirname "$file")"
         mkdir -p "$dest"
         cp "${npmWorkspace-.}/$file" "$dest"
-    done < <(@jq@ --raw-output '.[0].files | map(.path) | join("\n")' <<< "$(npm pack --json --dry-run ${npmWorkspace+--workspace=$npmWorkspace} $npmPackFlags "${npmPackFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}")")
+    done < <(@jq@ --raw-output '.[0].files | map(.path) | join("\n")' <<< "$(npm pack --json --dry-run --loglevel=warn --no-foreground-scripts ${npmWorkspace+--workspace=$npmWorkspace} $npmPackFlags "${npmPackFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}")")
 
     while IFS=" " read -ra bin; do
         mkdir -p "$out/bin"

